packages/devtools_app/lib/src/memory/memory_screen.dart

```diff
@@ -651,3 +1099,140 @@ class MemoryBodyState extends State<MemoryBody>
+    _fetch();
+  }
+
+  final MemoryBodyState memoryState;
+
+  final int index;
+
+  final int timestamp;
+
+  final _event = <String, Object>{};
+
+  final _extensionEvents = <Map<String, Object>>[];
+
+  Map<String, Object> get vmData => _vm;
+
+  final _vm = <String, Object>{};
+
+  Map<String, Object> get androidData => _android;
+
+  final _android = <String, Object>{};
+
+  Map<String, Object> toJson() {
+    return {
+      'index': index,
+      'timestamp': timestamp,
+      'prettyTimestamp': prettyTimestamp(timestamp),
+      'event': _event,
+      'vm': _vm,
+      'android': _android,
+    };
+  }
+
+  bool get hasSnapshot => _event.containsKey(snapshotDisplayName);
+  bool get hasAutoSnapshot => _event.containsKey(autoSnapshotDisplayName);
+  bool get hasMonitorStart => _event.containsKey(monitorStartDisplayName);
+  bool get hasMonitorReset => _event.containsKey(monitorResetDisplayName);
+  bool get hasExtensionEvents => _event.containsKey(extensionEventsDisplayName);
+  bool get hasManualGc => _event.containsKey(manualGCDisplayName);
+  bool get hasGc => _vm.containsKey(gcDisplayName);
+
+  int get extensionEventsLength => hasExtensionEvents
+      ? (_event[extensionEventsDisplayName] as List).length
+      : 0;
+
+  List<Map<String, Object>> get extensionEvents {
+    if (_extensionEvents.isEmpty)
+      _extensionEvents.addAll(_event[extensionEventsDisplayName]);
+    return _extensionEvents;
+  }
+
+  void _fetch() {
+    _event.clear();
+    _vm.clear();
+    _android.clear();
+
+    _fetchEventData(_event);
+    _fetchData(memoryState.vmChartController, _vm);
+    _fetchData(memoryState.androidChartController, _android);
+  }
+
+  void _fetchEventData(Map<String, Object> results) {
+    // Use the detailed extension events data stored in the memoryTimeline.
+    final eventInfo =
+        memoryState.controller.memoryTimeline.data[index].memoryEventInfo;
+
+    if (eventInfo.isEmpty) return;
+
+    if (eventInfo.isEventGC) results[manualGCDisplayName] = true;
+    if (eventInfo.isEventSnapshot) results[snapshotDisplayName] = true;
+    if (eventInfo.isEventSnapshotAuto) results[autoSnapshotDisplayName] = true;
+    if (eventInfo.isEventAllocationAccumulator) {
+      if (eventInfo.allocationAccumulator.isStart) {
+        results[monitorStartDisplayName] = true;
+      }
+      if (eventInfo.allocationAccumulator.isReset) {
+        results[monitorResetDisplayName] = true;
+      }
+    }
+
+    if (eventInfo.hasExtensionEvents) {
+      final List<Map<String, Object>> events = [];
+      for (ExtensionEvent event in eventInfo.extensionEvents.theEvents) {
+        if (event.customEventName != null) {
+          events.add(
+            {
+              'name': event.eventKind,
+              'custom': {
+                'name': event.customEventName,
+                'data': event.data,
+              },
+            },
+          );
+        } else {
+          events.add(
```

Comment:
nit: this all fits on one line. feel free to remove spurious trailing commas.

Reply:
Done.